articles/implementing-caching-in-nodejs-using-redis/index.md

```diff
@@ -0,0 +1,331 @@
+Caching refers to the process of storing data in a temporary location so that the data is accessed with minimal resources. Caching aims to reduce the cost of bandwidth of data sent over the network and the application's response time. Applications that implement caching are faster and user friendly.
+
+#### prerequisites.
+
+To follow along in this article, it is helpful to have the following:
+
+- [Node.js](https://nodejs.org/en/) installed on your computer.
+- Working knowledge of JavaScript.
+- Working knowledge of Node.js and [Express.js](https://expressjs.com/).
+- [Postman](https://www.postman.com/) installed on your computer.
+- Linux or macOS based operating system.
+
+### Overview
+
+- [Application overview](#application-overview)
+
+- [Setting up the application](#setting-up-the-application)
+
+- [Fetching data from recipe labs API](#fetching-data-from-recipe-labs-api)
+
+- [Installing redis](#installing-redis)
+
+- [Importing and configuring redis.](#importing-and-configuring-redis)
+
+- [Setting up redis](#setting-up-redis)
+
+- [Considerations while implementing caching](#considerations-while-implementing-caching)
+
+### Application overview.
+
+In this article, we will walk through the steps of implementing caching in a Node.js app using [redis](https://redis.io/) as our cache. We will implement a REST API using the Express.js framework. We will be making requests to [recipe labs api](http://www.recipepuppy.com/?ref=hackernoon.com). It is a public API for accessing recipes of different food items.
+
+### Setting up the application.
+
+To set up the application, clone this [Github repository](https://github.com/mwangiKibui/implementing-caching-in-nodejs). The repository has two directories, start, and final. We will be working on the start directory throughout the article but if you encounter an error feel free to check out the final directory.
+
+To install the necessary dependencies, in your cloned folder execute the following command:
+
+```bash
+cd ./start && npm install
+```
+
+Our main focus shall be implementing `fetchFoodItem` function in `controllers/recipe.js` file. It handles the logic for the `recipe/:foodItem` route.
+
+### Fetching data from recipe labs API.
+
+To fetch data from the API we:
+
+- Import `axios` module:
+
+```javascript
+const axios = require("axios");
+```
+
+- Modify the `fetchFoodItem` function as follows:
```

Comment:
It's important to point out that the `fetchFoodItem` function is located in `controllers/recipe.js`